### DISCUSSION TITLE
Trying parallel training for cpu version of deepmd-kit

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#1900] [Q&A] [answered]
I was trying to do parallel training for the CPU version of deepmd-kit to compare the performance b/w CPU single thread,multi-thread, and multi-node.

Command -- 
mpirun  --hostfile hostfile -np 4  dp train --mpi-log=workers input.json

hostfile Content --
localhost slots=6

Output --
![Screenshot 2022-09-08 at 3 26 11 PM](https://user-images.githubusercontent.com/10074710/189093341-7623c638-7d56-4e88-8ae8-0246a0cade2b.png)

I want to know why it is not running in parallel or if this feature is limited to the GPU version of deepmd-kit.. Usually, we are able to launch multiple processes for regular code via mpirun. If I want to train to CPU cluster with multi-cores, how should I go about it?

**Comment by njzjz** [comment]
Did you install horovod?

**Reply by AnuragKr** [reply]
No, I didn't install horovod, Do I need to install horovod to make mpirun work?

**Reply by njzjz** [reply]
Yes.

**Comment by AnuragKr** [comment, marked as answer]
Thanks, @njzjz. It worked after installing horovod.
